### DISCUSSION TITLE
How Can I Get Param In lua script when I Call CallByParam

### DISCUSSION THREAD

**Discussion by 2276282419** [yuin/gopher-lua#384] [Q&A]
- [ ] GopherLua is a Lua5.1 implementation. You should be familiar with Lua programming language. Have you read [Lua 5.1 reference manual](http://www.lua.org/manual/5.1/) carefully?
- [ ] GopherLua is a Lua5.1 implementation.  In Lua, to keep it simple, **it is more important to remove functionalities rather than to add functionalities unlike other languages** . If you are going to introduce some new cool functionalities into the GopherLua code base and the functionalities can be implemented by existing APIs, It should be implemented as a library. 

Please answer the following before submitting your issue:

1. What version of GopherLua are you using? : 
2. What version of Go are you using? : 
3. What operating system and processor architecture are you using? :
4. What did you do? :
5. What did you expect to see? :
6. What did you see instead? :

**Comment by 2276282419** [comment]
fn, err := L.LoadString(`
	  function hello(message_part)
	    print(message_part)
		return "hi"
	  end
	  hello("e")
	`)	
L.SetGlobal("hello", fn)
	if err := L.CallByParam(lua.P{
		Fn:      L.GetGlobal("hello"),
		NRet:    1,
		Protect: true,
	}, lua.LNumber(10)); err != nil {
		panic(err)
	}
I want get number 10 in hello function

**Comment by waschik** [comment]
The function LoadString returns the full string. So each time you call fn it redefines hello. In this case `DoString` might be easier:
```go
	err := L.DoString(`
          function hello(message_part)
            print(message_part)
	    return "hi"
          end
          hello("e")
        `)
	if err != nil {
		panic(err)
	}

	err = L.CallByParam(lua.P{
		Fn:      L.GetGlobal("hello"),
		NRet:    1,
		Protect: true,
	}, lua.LNumber(10))
	if err != nil {
		panic(err)
	}
```

Alternatively you could call fn to actually define hello.
```go
	fn, err := L.LoadString(`
          function hello(message_part)
            print(message_part)
	    return "hi"
          end
          hello("e")
        `)
	if err != nil {
		panic(err)
	}

	err = L.CallByParam(lua.P{
		Fn:      fn,
		NRet:    0,
		Protect: true,
	})
	if err != nil {
		panic(err)
	}

	err = L.CallByParam(lua.P{
		Fn:      L.GetGlobal("hello"),
		NRet:    1,
		Protect: true,
	}, lua.LNumber(10))
	if err != nil {
		panic(err)
	}
```

Output is in both cases:
```
e
10
```
